### ISSUE TITLE
Light.js from Johnny-Five doesn't work through j5-io

### ISSUE THREAD

**Issue by jhoughjr** [nebrius/raspi-io-core#10]
Light.js calls i2cWrite(address, mode)
but j5-io provides a different signature.
I'm not sure how to resolve this.
(I'm trying to use a bh1750 on an rpi over i2c.)
Any ideas?

**Comment by jhoughjr** [comment]
Doing more digging, Raspi IO seems to state i2cWrite should have a form that doesn't take a register.
Still investigating.

**Comment by nebrius** [comment]
Good thinking opening a new issue here

Reiterating from https://github.com/rwaldron/johnny-five/pull/1302, here's the signature for the method throwing the exception (written in TypeScript):

```typescript
  public i2cWrite(address: number, inBytes: number[]): void;
  public i2cWrite(address: number, register: number, inBytes: number[]): void;
```

Notice how the first signature doesn't take a register, so that's not the issue at play. It still seems to be some unknown third signature. @dtex did you get a chance to investigate further?

**Comment by nebrius** [comment]
@jhoughjr can you post the stack trace you get with this exception? This will help me to isolate the code in J5 a little better that's calling with an unknown signature.

**Comment by dtex** [comment]
I did not investigate much beyond [this comment](https://github.com/rwaldron/johnny-five/pull/1302#issuecomment-470239825). I did take a look at the unit tests and saw that the mocks do not match firmata functionality which is why the tests are passing.

I don't have a BH1750 so I can't test with real hardware and I've never worked on the light class so there is a very real chance that I'm missing something when reading the code and it should actually work just fine.

@jhoughjr Would you mind trying this and seeing if it works for you:

Change line 409 in /node_modules/johnny-five/lib/light.js from:
````
var mode = opts.mode || 0x10;
````
to:
````
var mode = [opts.mode] || [0x10];
````

**Comment by jhoughjr** [comment]
I'm traveling today but I can try it tomorrow.

**Comment by dtex** [comment]
Oh, I see how it works now. 

@nebrius There is an undocumented yet valid signature that is used throughout Johnny-Five.

````
public i2cWrite(address: number, register: number): void;
````

firmata.js handles it [here](https://github.com/firmata/firmata.js/blob/master/packages/firmata-io/lib/firmata.js#L1239-L1248)

I will add documentation over in firmata.js

**Comment by nebrius** [comment]
> public i2cWrite(address: number, register: number): void;

That's... _strange_ (why write if you're not going to write anything?), but ok, I'll implement it.

**Comment by nebrius** [comment]
I just released v3.1.0 of j5-io which includes the new signature. This should fix* your issue.

&ast; Technically this restores the old behavior, which _ignores_ calls with this signature. As implied above and now verified, it's _impossible_ for me to write to I2C without a value. I assume this call is happening for a reason though, so ignoring it may not produce the desired behavior, but it does restore the old behavior (which may have not been desired either).

**Comment by nebrius** [comment]
Closing due to inactivity. Feel free to reopen if you're still seeing this behavior.
